control-plane/cmd/post-install/kafka_channel_migrator.go

```diff
@@ -0,0 +1,291 @@
+	cont := ""
+	first := true
+	lsOpts := metav1.ListOptions{LabelSelector: labels.SelectorFromSet(kafkaChannelServiceLabels).String(), Continue: cont}
+	for first || cont != "" {
+		kafkaChannelServiceList, err := m.k8s.CoreV1().
+			Services(corev1.NamespaceAll).
+			List(ctx, lsOpts)
+		if err != nil {
+			return fmt.Errorf("failed to list KafkaChannel services: %w", err)
+		}
+		for _, svc := range kafkaChannelServiceList.Items {
+
+			patch := []byte(fmt.Sprintf(`[{"op":"replace", "path": "/spec/externalName", "value": "%s"}]`, newDispatcherExternalName))
+
+			logger.Infof("Patching service %s/%s with the patch: %s.", svc.Namespace, svc.Name, patch)
+
+			_, err := m.k8s.CoreV1().
+				Services(svc.Namespace).
+				Patch(ctx, svc.Name, types.JSONPatchType, patch, metav1.PatchOptions{})
+			if err != nil {
+				// just log and continue with other services
+				logger.Errorf("error while patching KafkaChannel service %s/%s: %w", svc.Namespace, svc.Name, err)
+			}
+		}
+		cont = kafkaChannelServiceList.Continue
+		first = false
+	}
+	return nil
+}
+
+func (m *kafkaChannelMigrator) migrateConfigmap(ctx context.Context, err error, logger *zap.SugaredLogger) error {
+	// consolidated configmap looks like this:
+	//
+	// apiVersion: v1
+	// data:
+	//   eventing-kafka: |
+	//     kafka:
+	//       brokers: my-cluster-kafka-bootstrap.kafka:9092
+	//       authSecretNamespace: my-namespace
+	//       authSecretName: my-secret
+	//   version: 1.0.0
+	// kind: ConfigMap
+	// metadata:
+	//   name: config-kafka
+	//   namespace: knative-eventing
+
+	// the new configmap should look like this:
+	//
+	// apiVersion: v1
+	// kind: ConfigMap
+	// metadata:
+	//   name: kafka-channel-config
+	//   namespace: knative-eventing
+	//  data:
+	//    bootstrap.servers: "my-cluster-kafka-bootstrap.kafka:9092"
+	//    auth.secret.ref.namespace: my-ns
+	//    auth.secret.ref.name: my-secret
+	// These 2 things don't have anything in old configmap. Defaults of them in the old channel are hardcoded.
+	//    default.topic.partitions: "10"
+	//    default.topic.replication.factor: "3"
+
+	// get the old configmap, extract these and set it in the new configmap:
+	// OLD CM PATH                                   NEW CM PATH
+	// ----------------------------                  ------------------------
+	// eventing-kafka/kafka/brokers              --> bootstrap.servers
+	// eventing-kafka/kafka/authSecretNamespace  --> auth.secret.ref.namespace
+	// eventing-kafka/kafka/authSecretName       --> auth.secret.ref.name
+	//
+
+	logger.Infof("Migrating configmap.")
+
+	oldcm, err := m.k8s.CoreV1().
+		ConfigMaps(system.Namespace()).
+		Get(ctx, OldConfigmapName, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		// configmap will be missing if we did the migration already
+		return fmt.Errorf("failed to get consolidated channel configmap for migration %s: %w", OldConfigmapName, err)
+	}
```

Comment:
We should return early without error if the ConfigMap `OldConfigmapName` is not found since `oldcm` is nil if we can't find it and there is nothing to migrate.